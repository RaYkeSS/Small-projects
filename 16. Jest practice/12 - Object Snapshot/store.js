import { combineReducers, configureStore } from "@reduxjs/toolkit";

import { addUser } from "./obj.js";
import usersReducer from "./obj.js";

const rootReducer = combineReducers({
  users: usersReducer,
});

export function setupStore(preloadedState) {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
}

const store = setupStore();
store.dispatch(addUser("Kevin"));
export const logo = store.getState();
