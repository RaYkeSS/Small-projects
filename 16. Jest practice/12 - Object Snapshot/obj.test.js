import { jest, expect, describe, test } from "@jest/globals";

import usersReducer, { addUser } from "./obj.js";
import { setupStore } from "./store.js";

describe("usersReducer", () => {
  test("initial test", () => {
    const store = setupStore();
    store.dispatch(addUser("Kevin"));
    expect(store.getState()).toEqual({
      users: ["Kevin"],
    });
  });

  test("should initialize as empty array", () => {
    const store = setupStore();
    expect(store.getState()).toEqual({
      users: [],
    });
  });
  describe("snapshots matching", () => {
    const store = setupStore();

    test("should append to state array", () => {
      // act
      store.dispatch(
        addUser({
          name: "Kyle Welch",
          handle: "kwelch",
        })
      );

      // assert
      expect(store.getState()).toMatchSnapshot();

      // act
      store.dispatch(
        addUser({
          name: "Jane Smith",
          handle: "jsmith",
        })
      );

      // assert
      expect(store.getState()).toMatchSnapshot();
    });

    test("should update when handle matches", () => {
      // act
      store.dispatch(
        addUser({
          name: "Kyle Welch",
          handle: "kwelch",
          role: "Test Driven Developer",
        })
      );

      // assert
      expect(store.getState()).toMatchSnapshot();
    });
  });
});
