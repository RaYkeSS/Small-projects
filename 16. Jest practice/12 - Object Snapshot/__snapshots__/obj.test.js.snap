// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usersReducer snapshots matching should append to state array 1`] = `
{
  "users": [
    {
      "handle": "kwelch",
      "name": "Kyle Welch",
    },
  ],
}
`;

exports[`usersReducer snapshots matching should append to state array 2`] = `
{
  "users": [
    {
      "handle": "kwelch",
      "name": "Kyle Welch",
    },
    {
      "handle": "jsmith",
      "name": "Jane Smith",
    },
  ],
}
`;

exports[`usersReducer snapshots matching should update when handle matches 1`] = `
{
  "users": [
    {
      "handle": "kwelch",
      "name": "Kyle Welch",
    },
    {
      "handle": "jsmith",
      "name": "Jane Smith",
    },
    {
      "handle": "kwelch",
      "name": "Kyle Welch",
      "role": "Test Driven Developer",
    },
  ],
}
`;
